### PR TITLE
Modification about prosemirro-model

### PR DESCRIPTION
**Pull request by herrdu**
1. add `findIndex` method in Fragment
2. modiy `slice` params in Node

Please fill in this template.

- [ ] modify `slice` method params; add `findIndex` method in `Fragment`

If changing an existing definition:
- [ ]  About fragment code: https://github.com/ProseMirror/prosemirror-model/blob/master/src/fragment.js
- [ ] About model code: https://github.com/ProseMirror/prosemirror-model/blob/master/src/node.js

### CODE REVIEW

File: types/prosemirror-model/index.d.ts [hunk 1]
```diff
@@ -157,6 +157,13 @@ export class Fragment<S extends Schema = any> {
      * object with two separate positions is returned.
      */
     findDiffEnd(other: Fragment<S>): { a: number; b: number } | null | undefined;
+    /**
+     * : (number, ?number) → {index: number, offset: number}
+     * Find the index and inner offset corresponding to a given relative
+     * position in this fragment. The result object will be reused
+     * (overwritten) the next time the function is called. (Not public.)
```
Comment (AnthonyWeston) [inline types/prosemirror-model/index.d.ts:164]: This method isn't public

File: types/prosemirror-model/index.d.ts [hunk 2]
```diff
@@ -597,7 +604,7 @@ declare class ProsemirrorNode<S extends Schema = any> {
      * Cut out the part of the document between the given positions, and
      * return it as a `Slice` object.
      */
-    slice(from: number, to?: number): Slice<S>;
+    slice(from: number, to?: number, includeParents?: boolean): Slice<S>;
```
Comment (AnthonyWeston) [inline types/prosemirror-model/index.d.ts:607]: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52466#issuecomment-823377973

**Review by AnthonyWeston — CHANGES_REQUESTED** [review]
As @Narretz mentioned in #52466, you can open an issue on prosemirror-model. Otherwise I'd recommend trying module augmentation as described [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52466#issuecomment-823770179)

File: types/prosemirror-model/index.d.ts (hunk 1 above)
Comment (herrdu) [reply types/prosemirror-model/index.d.ts:164]: I got an error .

image

File: types/prosemirror-model/index.d.ts [hunk 3]
```diff
@@ -157,6 +157,13 @@ export class Fragment<S extends Schema = any> {
      * object with two separate positions is returned.
      */
     findDiffEnd(other: Fragment<S>): { a: number; b: number } | null | undefined;
+    /**
+     * : (number, ?number) → {index: number, offset: number}
```
Comment (stof) [inline types/prosemirror-model/index.d.ts:161]: this should be removed

Comment (herrdu) [reply types/prosemirror-model/index.d.ts]: has fixed it

File: types/prosemirror-model/index.d.ts (hunk 1 above)
Comment (stof) [reply types/prosemirror-model/index.d.ts:164]: but if the method is not public, that error is expected: you should not be using it

Comment (herrdu) [reply types/prosemirror-model/index.d.ts:164]: Maybe  you should read it 。

https://discuss.prosemirror.net/t/prosemirror-code-with-typescript/3712/8

image

**Pull request closed without merging**